packages/design-system-web/src/components/geov-yasgui-data-validation/geov-yasgui-data-validation.tsx

```diff
@@ -0,0 +1,90 @@
+import { Component, Host, Prop, h, EventEmitter, Event } from '@stencil/core';
+import type { Parser } from '@triply/yasr';
+
+interface ExpectedKey {
+  name: string;
+  required: boolean;
+  datatype: string;
+  customValidator?: (val: Parser.BindingValue) => Set<string> | undefined;
+}
+/**
+ * The component has to validate input data and emit the validation result (valid/invalid).
```

Comment:
We need a more detailed description here. What are the inputs, what is the output?
Especially important is a thorough description of **the Event that is emitted and that needs to be reacted upon** - since this is not really visible in the story. What is the event's name (_validationCompleted_) what is its value?

Reply:
I hope the additional texts are ok